pmd-core/src/main/java/net/sourceforge/pmd/lang/metrics/internal/DummyMetricMemoizer.java

```diff
@@ -0,0 +1,44 @@
+/**
+ * BSD-style license; for more info see http://pmd.sourceforge.net/license.html
+ */
+
+package net.sourceforge.pmd.lang.metrics.internal;
+
+import net.sourceforge.pmd.lang.ast.Node;
+import net.sourceforge.pmd.lang.metrics.MetricMemoizer;
+import net.sourceforge.pmd.lang.metrics.ParameterizedMetricKey;
+
+
+/**
+ * Memoizes nothing.
+ *
+ * @author Clément Fournier
+ * @since 7.0.0
+ */
+public final class DummyMetricMemoizer<N extends Node> implements MetricMemoizer<N> {
```

Comment:
I'd rename this as `NoopMetricMemoizer`.

For one, a dummy is tipically a plceholder, more related to testing than an actual implementation. This is a very real implementation that happens to do nothing.

Moreover, this would be consistent with other classes in the project such as `NoopAnalysisCache`

Reply:
Good idea, will do!